documentation.md

```diff
@@ -0,0 +1,70 @@
+---
+title: Documentation Strategy
+description: 
+published: true
+date: 2023-04-10T00:00:00.000Z
+tags: 
+editor: markdown
+dateCreated: 2023-04-10T00:00:00.000Z
+---
+
+This page describes our various documentation sources and seeks to eliminate gray area between them.
+
+## Docs site
+
+- Source: https://github.com/centerofci/mathesar/tree/master/docs
+- Published: https://docs.mathesar.org/
+- Scope:
+    - **User docs** (we don't have this yet, but we wil eventually)
+    - **Administrator docs** (for people installing/upgrading Mathesar)
+- How do edit:
+    1. Follow instructions within README to preview docs content locally using mkdocs.
+    1. Base your edits on the correct branch:
+        - Target `master` if you have an important fix which needs to be published for the currently-released version of Mathesar.
+        - Target `develop` if you are adding/updating documentation along with yet-to-be-released changes to the product.
+    1. Make a PR against the repo with your changes.
+- Notes:
+    - The docs site is published from the `master` branch. This is important because we want to ensure that it reflects the latest *released* version of Mathesar so that docs readers who are installing or using Mathesar don't see content before it's actually applicable.
+
+## Markdown near code
+
+- Source: Bespoke `.md` files scattered within https://github.com/centerofci/mathesar/
+- Published: *(nowhere, other than the GitHub web interface)*
+- Scope:
+    - **Developer docs**
+    - **Contributor guidelines**
+    - **Code standards**
+    - **Architectural overviews**
+    - **Module documentation**
+    - (basically, anything about the *code itself*)
+- How do edit:
+    1. Make a PR against the `develop` branch.
+- Tips:
+    - Put documentation close to the code that it documents.
+    - Use relative hyperlinks like `[Foo](./bar/foo.md)` to cross-reference other markdown files.
+    - Ensure that content is discoverable by linking to it from other places that readers are likely to look.
```

Comment:
@seancolsen I'd like to see more specific guidelines about this. This point describes a good high-level goal, but it doesn't provide enough information for someone working on the documentation to know what counts as discoverable or where to link things.

Reply:
Can we handle this later if it becomes a problem? I don't want to get hung up on providing this level of detail right now. I don't anticipate that adding any detail here will meaningfully improve discoverability in the future. After we get this PR merged, I'm planning to make a rather larger PR on the mathesar repo where I move in a bunch of content from the wiki and move some other docs around. Then I have several pieces of new dev docs content to add. I'm waiting on hammering out this docs strategy before I can get going on that. I'd rather work on the actual documentation than the documentation about the documentation. In the process of doing all that docs work, I'll be thinking about discoverability the whole time and working to optimize for it. I want you to trust that I'll do a good job with discoverability as I'm re-organizing content. The process of figuring out exactly how each piece of content will link to each other piece of content will be much more efficient if I can perform it during that reorganization process. If I have to list a bunch of links and cross references ahead of time right now, it's going to take extra time and just not be very accurate because I won't necessarily think of everything ahead of time.

Reply:
> I want you to trust that I'll do a good job with discoverability as I'm re-organizing content. 

I do trust that you will do a good job with discoverability – we talked about it in detail during our call. However, the purpose of having documentation standards is for the _entire team_ to be on the same page. If you were someone on the team or an external contributor that wasn't on the call that we had, and were just relying on this document to understand Mathesar's standards for documentation, there isn't enough information to do so.

As it stands, this document doesn't really add anything to what we discussed on our call. If the purpose is just for _you_ to move ahead with documentation work, we don't need this page at all. When I asked for documentation standards as the next step, I meant that I'd like a page that gives _everyone_ enough information to meaningfully contribute to documentation.

"Ensure that content is discoverable by linking to it from other places that readers are likely to look." is full of assumptions – where you think readers are likely to look is different from what others might think, and what you think might be enough or too many links might be not enough to others. 

>  If I have to list a bunch of links and cross references ahead of time right now, it's going to take extra time and just not be very accurate because I won't necessarily think of everything ahead of time.

If you cannot meaningfully address discoverability right now, then please do the actual documentation work first and then update this page. You can add a note that says you'll update the page with more details. But please make sure to allocate time to do that.